File: 백준/Silver/15904. UCPC는 무엇의 약자일까？/UCPC는 무엇의 약자일까？.py
import sys
input = sys.stdin.readline

string = input().rstrip()
now = 0
for s in string:
    if now == 0 and s == 'U': now = 1
    elif now == 1 and s == 'C': now = 2
    elif now == 2 and s == 'P': now = 3
    elif now == 3 and s == 'C': now = 4

if now == 4: print('I love UCPC')
else: print('I hate UCPC')

File: 백준/Silver/15904. UCPC는 무엇의 약자일까？/UCPC는 무엇의 약자일까？.c
#include <stdio.h>
#include <string.h>

char str[1005];
char search[5] = "UCPC";
int main(void) {
    int now = 0;
    gets(str);
    for (int i = 0; i < strlen(str); i++) {
        if (now < 4 && str[i] == search[now]) now += 1;
    }

    printf("%s", (now == 4) ? "I love UCPC" : "I hate UCPC");
    return 0;
}